dali/base/dafdesc.cpp

```diff
@@ -672,7 +672,7 @@ protected: friend class CFileDescriptor;
     {
         partIndex = idx;
         ismulti = false;
-        if (!isEmptyPTree(pt)) {
```

Comment:
Not sure what the error this was fixing is, but this appears to make the code worse. Why was it needed?

Reply:
Russ asked me about this one, the error was:
```
[ 61%] Building CXX object dali/base/CMakeFiles/dalibase.dir/danqs.cpp.o
In constructor ‘CPartDescriptor::CPartDescriptor(CFileDescriptorBase&, unsigned int, IPropertyTree*)’,
    inlined from ‘virtual void CFileDescriptor::setNumParts(unsigned int)’ at /mnt/e/HPCC/HPCC-Platform/dali/base/dafdesc.cpp:1977:75:
/mnt/e/HPCC/HPCC-Platform/dali/base/dafdesc.cpp:676:31: error: ‘this’ pointer is null [-Werror=nonnull]
  676 |             if (pt->getPropInt("@num",idx+1)-1!=idx)
      |                 ~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/mnt/e/HPCC/HPCC-Platform/dali/base/dafdesc.cpp:678:29: error: ‘this’ pointer is null [-Werror=nonnull]
  678 |             overridename.set(pt->queryProp("@name"));
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
```

I suggested making it if (pt && isEmptyPTree(pt)) as a test.

But it appears it's a false positive from the compiler, because the !isEmptyTree should already checks pt for null

Reply:
Compiler can't know that isEmptyPTree() checks for null. I guess the issue is that the compiler DOES see some calls that pass in pt=null.

It is a false positive, but adding the check may be a reasonable approach in this case.

Reply:
Alternatively, adding two constructors - one for the case where NULL is passed and one fo the other case, might be a cleaner fix